tests/Integration/Behaviour/Features/Scenario/Product/duplicate_product.feature

```diff
@@ -0,0 +1,158 @@
+      | link_rewrite     | en-US:smart-sunglasses ;fr-FR:lunettes-de-soleil   |
+      | redirect_type    | 301-product                                        |
+      | redirect_target  | product2                                           |
+    And carrier carrier1 named "ecoCarrier" exists
+    And carrier carrier2 named "Fast carry" exists
+    And I update product product1 shipping information with following values:
+      | width                            | 10.5                                                  |
+      | height                           | 6                                                     |
+      | depth                            | 7                                                     |
+      | weight                           | 0.5                                                   |
+      | additional_shipping_cost         | 12                                                    |
+      | delivery time notes type         | specific                                              |
+      | delivery time in stock notes     | en-US:product in stock ;fr-FR:en stock                |
+      | delivery time out of stock notes | en-US:product out of stock ;fr-FR:En rupture de stock |
+      | carriers                         | [carrier1,carrier2]                                   |
+    And I add new supplier supplier1 with following properties:
+      | name             | my supplier 1            |
+      | address          | Donelaicio st. 1         |
+      | city             | Kaunas                   |
+      | country          | Lithuania                |
+      | enabled          | true                     |
+      | description      | en-US:just a supplier    |
+      | meta title       | en-US:my supplier nr one |
+      | meta description | en-US:                   |
+      | meta keywords    | en-US:sup,1              |
+      | shops            | [shop1]                  |
+    And I set product product1 default supplier to supplier1 and following suppliers:
+      | reference         | supplier reference | product supplier reference     | currency | price tax excluded |
+      | product1supplier1 | supplier1          | my first supplier for product1 | USD      | 10                 |
+    And I set following related products to product product1:
+      | product2 |
+    And I update product product1 with following customization fields:
+      | reference    | type | name                                                                       | is required |
+      | customField1 | text | en-US:text on top of left lense ;fr-FR:texte en haut de la lentille gauche | true        |
+    And I add new attachment "att1" with following properties:
+      | description | en-US:puffin photo nr1 ;fr-FR:macareux |
+      | name        | en-US:puffin ;fr-FR:macareux           |
+      | file_name   | app_icon.png                           |
+    And I associate attachment "att1" with product product1
```

Comment:
I understand and agree with your concern. It does not only apply to this test, but for others too (data initiation hides actual test purpose), but the problem is that whatever is placed to `background` is called before each scenario. So Background can only contain assertions, whilst here we have actual insertions...
Maybe we could come up with an idea of how to "hide" these scnearios that in theory belongs to background ?

Reply:
Yes but I don't think it's a problem to insert data in the background These are the datas required to then perform your real checks We do that in many other behat tests (like creating two languages, currencies, shops, ...)

Reply:
once again
> the problem is that whatever is placed to background is called before **each** scenario

that means each time it will create new product with new id. Then the same reference does contain different product id and all fails... (I've been there before).

Reply:
ok as you explained in slack its ok that background is being executed before each scenario (it ensures that scenarios are not coupled). I might have been coupling scenarios too much or whataver (that didn't work out well for me in past). But I'll try to do it here.